src/Tribe/Aggregator/Record/Queue.php

```diff
@@ -249,45 +260,44 @@ public function process( $batch_size = null ) {
 			return $this;
 		}
 
-		if ( $this->is_fetching() ) {
-			$data = $this->record->prep_import_data();
+		$this->has_lock = $this->acquire_lock();
 
-			if (
-				'fetch' === $data
-				|| ! is_array( $data )
-				|| is_wp_error( $data )
-			) {
-				return $this->activity();
-			}
+		if ( $this->has_lock ) {
+			if ( $this->is_fetching() ) {
+				$data = $this->record->prep_import_data();
 
-			$this->init_queue( $data );
-			$this->save();
-		}
+				if (
+					'fetch' === $data
+					|| ! is_array( $data )
+					|| is_wp_error( $data )
+				) {
+					return $this->activity();
+				}
 
-		// Every time we are about to process we reset the next var
-		$this->next = array();
+				$this->init_queue( $data );
+				$this->save();
+			}
 
-		if ( ! $batch_size ) {
-			$batch_size = apply_filters( 'tribe_aggregator_batch_size', Tribe__Events__Aggregator__Record__Queue_Processor::$batch_size );
-		}
 
-		for ( $i = 0; $i < $batch_size; $i++ ) {
-			if ( 0 === count( $this->items ) ) {
-				break;
+			if ( ! $batch_size ) {
+				$batch_size = apply_filters( 'tribe_aggregator_batch_size', Tribe__Events__Aggregator__Record__Queue_Processor::$batch_size );
```

Comment:
add dockblock for this filter

_note: since this is just merging branches, can ignore this for now if it will slow things down unnecessarily, but should ultimately be docblocked_